source/_posts/2025-05-29-ai-assisted-coding.markdown

```diff
@@ -0,0 +1,188 @@
+---
+title: AI-assisted coding for teams that can't get away with vibes
+kind: article
+author: Atharva Raykar
+created_at: 2025-05-29 00:00:00 UTC
+layout: post
+---
+*Status: Living document based on production experience*
+*Last updated: 5-Jun-2025*
+
+AI should be adopted by serious engineering teams that want to build thoughtful, well-crafted products. This requires skillful usage of these tools. Our obsession with building high-quality software for over a decade has driven us to figure out how this new way of building can result in better products.
+
+This living playbook is based on our experience working with AI tools in the messy trenches of production software, where no one can afford to get away with vibes. I hope other teams can learn and benefit from our findings.
+
+## Why build with AI?
+
+**Building with AI is fast**. With the right kind of usage, I have indeed seen massive improvements in velocity for shipping new features. The velocity improvements are not evenly spread. In our experience, the smaller straightforward tasks benefit massively more from AI compared to bigger, gnarlier work such as designing a good architecture for a system. But the speed-up in the smaller ongoing engineering work helps teams allocate more time and focus on harder problems. Moreover, when velocity increases at an individual level, you can eliminate coordination overhead with leaner teams, which can unlock even more velocity.
+
+This velocity is important, because when harnessed correctly, it allows teams to tighten feedback loops with users faster and make the product better. Any team that does not adopt AI tools effectively will be outcompeted by teams that ship faster using them.
+
+The hype around the velocity has led to some people and teams hastily adopting and pushing AI tools, and finding themselves disappointed by the output and quality. There is a good reason for this. AI tools are sophisticated tools that are frankly, difficult to use and unintuitive. Hold it wrong, and you can generate underwhelming results, worse still, slow down your velocity by drowning your project in slop and technical debt. I have seen this play out time after time.
+
+## AI is a multiplier
```

Comment:
Think if it starts with the `What helps the human helps the AI` heading it'll be punchier. Could just drop these 2 sections imo.

Reply:
The prompt example is nice, but it might be better to link to a simonw post or something more canonical here.